core/src/main/java/org/testng/ITestResult.java

```diff
@@ -108,6 +108,14 @@
     return Collections.emptyList();
   }
 
+  /**
+   * @return - A unique id for the current JVM that represents a unique way of identifying a
+   * specific test method's result.
+   */
+  default String id() {
+    return null;
```

Comment:
I think throwing an exception with a comprehensive message is better than an NPE later in `Reporter`.

Reply:
@juherr - Are you suggesting that someone outside of TestNG is going to try and implement the `ITestResult` interface and that would somehow land up with the TestNG codebase ? I am not sure how that can happen. Can you please help me understand the use-case when that would happen ?

Reply:
> I am not sure how that can happen.

Agree! No default method, just a declaration then ;)

Reply:
Fixed. I have also rebased off of master and fixed merge conflicts and pushed in the amended commit. Please help take a look.